mitxgraders-js/MJxPrep.test.js

```diff
@@ -105,6 +105,12 @@ describe('preProcessEqn', () => {
     const eqn = 'cross(x) + cross(a + b, c) + y'
     expect(preProcessEqn(eqn)).toBe('cross(x) + {:(a + b) times c:} + y')
   } )
+
+  it('wraps primed expressions in unbreakable invisible brackets', () => {
+    const expr = "f''(x)/f(x) + g_a'(y) "
+    expect(preProcessEqn(expr)).toBe("{:f''(x):}/f(x) + {:g_a'(y):}")
```

Comment:
I just added this test case. Based on the behavior of [http://asciimath.org/](http://asciimath.org/), I am pretty sure this test does describe the desired behavior.

Currently this test fails:
```
Expected: "{:f''(x):}/f(x) + {:g_a'(y):}"
Received: "{:f'':}(x)/f(x) + g_{:a':}(y) "
```
(I just noticed the extra space at the end in definition of `expr`, but that's not the main issue.)

Do you want to try and fix it? I can also work on it, but probably not till Wednesday or Thursday.

Reply:
Oh, I see. You're saying this test fails because of a different issue?

Re:
> Which actually brings up a different point: should functions be wrapped, so that something like p(x)/2 doesn't look terrible?

Yes, that sounds reasonable. And it might make a whole bunch of our existing preprocessing redundant?

Basically:

- anything of the for `[non-operator characters](...)` must be a function call
- and so should be wrapped

right?

Reply:
Yeah, I think that's the case. Would probably be good to have all functions be wrapped (I don't see how this could break anything?), which would look after the prime problem for functions. However, there's still a prime problem for variables. We can modify the regex to use lookahead to see if the variable is proceeded by a (, in which case it's a function and can be ignored.

Why don't I modify this to fix the variable prime problem only, and leave you to fix the function problem in a separate PR?

Reply:
> Why don't I modify this to fix the variable prime problem only, and leave you to fix the function problem in a separate PR?

That sounds good. I'm not sure if you need to modify anything other than my broken test :)

Reply:
I need to detect that I'm not in a function ;-)